src/export/integers.rs

```diff
@@ -0,0 +1,36 @@
+use super::*;
+
+macro_rules! write_to_le_bytes {
+    ($s:ident) => {
+        /// Implementation detail
+        pub fn $s(b: &$s) {
+            write(&b.to_le_bytes())
+        }
+    };
+}
+
+/// Implementation detail
+pub fn u8(b: &u8) {
+    write(&[*b])
+}
+
+write_to_le_bytes!(u16);
+write_to_le_bytes!(u32);
+write_to_le_bytes!(u64);
+write_to_le_bytes!(u128);
+
+/// Implementation detail
+pub fn usize(b: &usize) {
+    write(&(*b as u32).to_le_bytes())
+}
+
+write_to_le_bytes!(i8);
+write_to_le_bytes!(i16);
+write_to_le_bytes!(i32);
+write_to_le_bytes!(i64);
+write_to_le_bytes!(i128);
```

Comment:
This would be simpler if the macro took a list of types, then you only have to call it once